code/controllers/subsystems/pai.dm

```diff
@@ -136,116 +124,28 @@
-
-		<table>
-			<tr class="d0">
-				<th rowspan="2"><a href='byond://?src=\ref[src];option=name;new=1;allow_submit=[allowSubmit];candidate=\ref[candidate]'>Name</a>:</th>
-				<td class="desc">[candidate.name]&nbsp;</td>
-			</tr>
-			<tr class="d1">
-				<td>What you plan to call yourself. Suggestions: Any character name you would choose for a station character OR an AI.</td>
-			</tr>
-			<tr class="d0">
-				<th rowspan="2"><a href='byond://?src=\ref[src];option=desc;new=1;allow_submit=[allowSubmit];candidate=\ref[candidate]'>Description</a>:</th>
-				<td class="desc">[candidate.description]&nbsp;</td>
-			</tr>
-			<tr class="d1">
-				<td>What sort of pAI you typically play; your mannerisms, your quirks, etc. This can be as sparse or as detailed as you like.</td>
-			</tr>
-			<tr class="d0">
-				<th rowspan="2"><a href='byond://?src=\ref[src];option=role;new=1;allow_submit=[allowSubmit];candidate=\ref[candidate]'>Preferred Role</a>:</th>
-				<td class="desc">[candidate.role]&nbsp;</td>
-			</tr>
-			<tr class="d1">
-				<td>Do you like to partner with sneaky social ninjas? Like to help security hunt down thugs? Enjoy watching an engineer's back while he saves the station yet again? This doesn't have to be limited to just station jobs. Pretty much any general descriptor for what you'd like to be doing works here.</td>
-			</tr>
-			<tr class="d0">
-				<th rowspan="2"><a href='byond://?src=\ref[src];option=ooc;new=1;allow_submit=[allowSubmit];candidate=\ref[candidate]'>OOC Comments</a>:</th>
-				<td class="desc">[candidate.comments]&nbsp;</td>
-			</tr>
-			<tr class="d1">
-				<td>Anything you'd like to address specifically to the player reading this in an OOC manner. \"I prefer more serious RP.\", \"I'm still learning the interface!\", etc. Feel free to leave this blank if you want.</td>
-			</tr>
-		</table>
-		<br>
-	"}
-	if(allowSubmit)
-		dat += {"
-			<table>
-				<td class="button"><a href='byond://?src=\ref[src];option=submit;new=1;candidate=\ref[candidate]' class="button"><b><font size="4px">Submit Personality</font></b></a></td>
-			</table><br>
-			"}
-	dat += {"
-	<body>
-	"}
-
-	M << browse(dat, "window=paiRecruit;size=580x580;")
-
-// TODO: Refactor the spaghetti above into NanoUI.
+		var/pai = M.client.prefs.pai
+		if(pai["name"])
+			candidate.name = sanitizeSafe(pai["name"], MAX_NAME_LEN)
+		if(pai["description"])
+			candidate.description = sanitize(pai["description"])
+		if(pai["role"])
+			candidate.role = sanitize(pai["role"])
+		if(pai["comments"])
+			candidate.comments = sanitize(pai["comments"])
+
+	var/datum/vueui/ui = SSvueui.get_open_ui(M, src)
+	if(!ui)
+		ui = new(M, src, "misc-pai-recruit", 580, 590, "pAI Personality Configuration", list(
+			"name" = candidate.name || "",
+			"description" = candidate.description || "",
+			"role" = candidate.role || "",
+			"comments" = candidate.comments || ""
+		), nstate = interactive_state)
```

Comment:
Could push this parenthesis up for better readability? Not really a blocking concern though, up to you.